mlflow/projects/__init__.py

```diff
@@ -67,12 +70,19 @@ def _resolve_experiment_id(experiment_name=None, experiment_id=None):
     if experiment_name and experiment_id:
         raise MlflowException("Specify only one of 'experiment_name' or 'experiment_id'.")
 
-    exp_id = experiment_id
+    if experiment_id:
+        return experiment_id
+
     if experiment_name:
         client = tracking.MlflowClient()
-        exp_id = client.get_experiment_by_name(experiment_name).experiment_id
-    exp_id = exp_id or _get_experiment_id()
-    return exp_id
+        exp = client.get_experiment_by_name(experiment_name)
+        if exp:
+            return exp.experiment_id
+        else:
+            print("INFO: '{}' does not exist. Creating a new experiment".format(experiment_name))
+            return client.create_experiment(experiment_name)
```

Comment:
you want to return the experiment ID right?

Reply:
It'd be worth adding some simple unit tests for this function so we can catch things like this. Maybe in https://github.com/mlflow/mlflow/blob/master/tests/projects/test_projects_cli.py. `client.create_experiment` can be mocked.

Reply:
Will do! I think `client.create_experiment` returns the created ID, but we'll find out for sure when I do the tests :)

Reply:
Ah my bad - you're right it returns the ID. Probably still a good idea to add tests to ensure the different cases work though. Thank you!